pkgs/by-name/al/alist/package.nix

```diff
@@ -0,0 +1,83 @@
+{
+  lib,
+  alist,
+  buildGoModule,
+  fetchFromGitHub,
+  fetchurl,
+  fuse,
+  go,
+  stdenv,
+  installShellFiles,
+  testers,
+}:
+buildGoModule rec {
+  pname = "alist";
+  version = "3.38.0";
+
+  src = fetchFromGitHub {
+    owner = "alist-org";
+    repo = "alist";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-1nqveTEYrN9kYAOMTbLPw0MFNNNbFOqej+d+aTLAZ1o=";
+  };
+  web = fetchurl {
+    url = "https://github.com/alist-org/alist-web/releases/download/${version}/dist.tar.gz";
+    hash = "sha256-jHbWhjvHfgtdocuuELbOwrMzL8tOQfBVdH9MxasEwGo=";
+  };
+
+  CGO_ENABLED = 1;
+
+  vendorHash = "sha256-qSbvb2/y5rdS/OCutNEcRDUQBCAgNudux8XDnY9TRSo=";
+
+  buildInputs = [ fuse ];
+
+  tags = [ "jsoniter" ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X \"github.com/alist-org/alist/v3/internal/conf.GitAuthor=Xhofe <i@nn.ci>\""
+    # time should not be used as the input for nix, use "Nix" as a placeholder instead.
+    "-X github.com/alist-org/alist/v3/internal/conf.BuiltAt=Nix"
+    "-X github.com/alist-org/alist/v3/internal/conf.GoVersion=${go.version}"
+    "-X github.com/alist-org/alist/v3/internal/conf.GitCommit=${version}"
+    "-X github.com/alist-org/alist/v3/internal/conf.Version=${version}"
+    "-X github.com/alist-org/alist/v3/internal/conf.WebVersion=${version}"
+  ];
+
+  preConfigure = ''
+    # build sandbox do not provide network
+    rm pkg/aria2/rpc/client_test.go pkg/aria2/rpc/call_test.go
+
+    # use matched web files
+    rm -rf public/dist
+    cp ${web} dist.tar.gz
+    tar -xzf dist.tar.gz
+    mv -f dist public
+  '';
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    installShellCompletion --cmd alist \
+      --bash <($out/bin/alist completion bash) \
+      --fish <($out/bin/alist completion fish) \
+      --zsh <($out/bin/alist completion zsh)
+  '';
+
+  passthru.tests = {
+    version = testers.testVersion {
```

Comment:
[`versionCheckHook`][1] is [preferred][2] over `testers.testVersion`.  Besides reasons mentioned in the doc, `testers.testVersion` cannot handle `overrideAttrs` here because `buildGoModule` does not support the `finalAttrs` pattern yet.

```nix
  doInstallCheck = true;
  versionCheckProgramArg = "version";
  nativeInstallCheckInputs = [
    versionCheckHook
  ];
```

[1]: https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook
[2]: https://nixos.org/manual/nixpkgs/unstable/#var-passthru-tests-packages